### ISSUE TITLE
2.0.0 -> 3.1.1 performance degradation

### ISSUE THREAD

**Issue by bugproof** [dezhidki/Tommy#33]
https://github.com/bugproof/TomlLibrariesBenchmark

**Comment by dezhidki** [comment]
Hi!

Thanks for bringing this up; this is quite a significant slowdown. After some basic profiling, it seems that the change to support TOML 1.0.0 caused a regression in date parsing performance:

![kuva](https://user-images.githubusercontent.com/5202606/148285009-fe1654d3-b62f-4040-825f-40a7d27c965f.png)

The issue is likely the try-catch snippet which essentially causes unnecessary operations:

https://github.com/dezhidki/Tommy/blob/91e8fff297e0c5cc2885ae34f36e5394c553cddd/Tommy/Tommy.cs#L2011-L2018

Changing it back to original `TryParse` implementation indeed fixes the flamegraph:

![kuva](https://user-images.githubusercontent.com/5202606/148285375-ab79901c-37dc-4a64-b1b3-9e4dcab01077.png)

Of course, supporting full TOML 1.0.0 spec causes a slight performance degradation that I cannot fully fix. 
I pushed the fixed code right now (https://github.com/dezhidki/Tommy/commit/ecac69d7d8ffa7cbf6fa50620ac1b2ace6671396, no release yet) if you feel like giving it a test. Here's the library if you don't want to compile it from the source:

[Tommy.zip](https://github.com/dezhidki/Tommy/files/7817652/Tommy.zip)

 I don't know about your testing setup, but on mine:

```
BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19044.1415 (21H2)
AMD Ryzen 5 3600, 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.100
  [Host]     : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT
  DefaultJob : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT
```

The fixed code now runs in `20.54 us`, which is reasonable given the results with 2.0.0.

**Comment by bugproof** [comment]
Quite a big difference after replacing exceptions with `TryParse`. I will update the benchmark results when new version will be on NuGet.
